processor/src/main/java/org/mapstruct/ap/internal/model/assignment/AdderWrapper.java

```diff
@@ -61,7 +61,7 @@ public AdderWrapper( Assignment decoratedAssignment, List<Type> thrownTypesToExc
     public Set<Type> getImportTypes() {
         Set<Type> imported = new HashSet<Type>();
         imported.addAll( super.getImportTypes() );
-        imported.add( getSourceType() );
+        imported.add( getSourceType().getTypeParameters().get( 0 ) );
```

Comment:
So.. to answer the question.. the import is needed for the local var in the for loop.